mesh/reward_test.go

```diff
@@ -211,57 +211,85 @@ func TestMesh_updateStateWithLayer(t *testing.T) {
 	mesh, atxDB := getMeshWithMapState(t, "t1", s)
 	defer mesh.Close()
 
-	for i := 1; i <= numOfLayers; i++ {
-		createLayer(t, mesh, types.NewLayerID(uint32(i)), numOfBlocks, maxTxs, atxDB)
-		l, err := mesh.GetLayer(types.NewLayerID(uint32(i)))
-		assert.NoError(t, err)
-		mesh.ValidateLayer(l)
+	startLayer := types.GetEffectiveGenesis()
+
+	for i := 0; i < numOfLayers; i++ {
+		layerID := startLayer.Add(uint32(i))
+		createLayer(t, mesh, layerID, numOfBlocks, maxTxs, atxDB)
+		l, err := mesh.GetLayer(layerID)
+		require.NoError(t, err)
+		mesh.ValidateLayer(context.TODO(), l)
 	}
 
 	s2 := &MockMapState{Rewards: make(map[types.Address]*big.Int)}
 	mesh2, atxDB2 := getMeshWithMapState(t, "t2", s2)
 
-	// this should be played until numOfLayers -1 if we want to compare states
-	for i := 1; i <= numOfLayers-1; i++ {
-		blockIds := copyLayer(t, mesh, mesh2, atxDB2, types.NewLayerID(uint32(i)))
-		mesh2.HandleValidatedLayer(context.TODO(), types.NewLayerID(uint32(i)), blockIds)
+	// this should be played until numOfLayers-1 if we want to compare states
+	for i := 0; i < numOfLayers-1; i++ {
+		layerID := startLayer.Add(uint32(i))
+		blockIds := copyLayer(t, mesh, mesh2, atxDB2, layerID)
+		mesh2.HandleValidatedLayer(context.TODO(), layerID, blockIds)
 	}
+
 	// test states are the same when one input is from tortoise and the other from hare
-	assert.Equal(t, s.Txs, s2.Txs)
+	require.NotEqual(t, s.Txs, s2.Txs)
```

Comment:
@lrettig 
is this correct? if we want the same result, it should be equal right?

Reply:
the comment is a little misleading (see identical comment a few lines down), but the change is intentional and this should be correct. all layers _except_ the latest layer have been applied here, so the state should not be the same. then the latest layer is applied, and then the state should be the same. that's what this is testing.